### DISCUSSION TITLE
Can you provide machine-readable release data?

### DISCUSSION THREAD

**Discussion by joschi** [microsoft/openjdk#18] [Q&A] [answered]
Projects such as [Disco API](https://github.com/foojay2020/discoapi), [Java Metadata API](https://joschi.github.io/java-metadata/), and @raphw's [Overview of OpenJDK distributions](https://rafael.codes/openjdk/) provide a comprehensive overview over the JDK ecosystem and are being used as upstream data sources for various other projects, for example version managers such as [asdf-java](https://github.com/halcyon/asdf-java/).

These projects rely on accurate and readily available data about JDK releases of each vendor. While there are some "model students" such as [AdoptOpenJDK with their API](https://api.adoptopenjdk.net/) and Azul with their [Azul Zulu OpenJDK Discovery API](https://www.azul.com/downloads/zulu-community/api/), many JDK vendors do not provide such information.

Being derived from AdoptOpenJDK code, the Microsoft OpenJDK distribution could surely provide better metadata about their JDK releases than a [prerendered HTML page](https://www.microsoft.com/openjdk) this information would have to (incorrectly and with unnecessary effort) be scraped.

References:
- https://twitter.com/joschi83/status/1379510869579730945
- https://forums.aws.amazon.com/thread.jspa?threadID=335202
- https://mail.openjdk.java.net/pipermail/web-discuss/2021-February/000586.html

**Comment by karianna** [comment, marked as answer]
Hi @joschi -  The AdoptOpenJDK author happens to work in our group and so we're looking into this, but no promises yet!

**Reply by joschi** [reply]
> The AdoptOpenJDK author happens to work in our group and so we're looking into this, but no promises yet!

If the "full-fledged" version is not going to happen or will take some time to consider and implement, then a scraping-friendly website would also work. 😉

**Reply by karianna** [reply]
We have added some basic data for the Eclipse Adoptium Marketplace.  We may consider further API work.  If that is desirable please do open an issue!

**Comment by d3r3kk** [comment]
@joschi with the Microsoft Build of OpenJDK now present on the Eclipse-Adoptium marketplace, is a static JSON file still warranted?
